update_repo_properties.py

```diff
@@ -0,0 +1,240 @@
+            repo_type=RepoType(source["repo_type"]),
+            status_checks=cls.CHECK_MAP[source["status_checks"]],
+            **bool_flags
+        )
+
+
+def str_to_bool(val: str):
+    return val.lower() in ('y', 'yes', 'true')
+
+
+def parse_repo_list_properties(repo_data_path: str) -> List[RepoProperty]:
+    data_path = Path(repo_data_path)
+    if not data_path.exists:
+        print('repo data file does not exist')
+        return
+    repo_data = []
+    # Deal with extra columns
+    with open(data_path, 'r') as csvfile:
+        csv_reader = csv.DictReader(csvfile)
+        for row in csv_reader:
+            data_row = RepoProperty.from_dict(row)
+            repo_data.append(data_row)
+
+    return repo_data
+
+
+def main(
+    owner: str = "pcdshub",
+    repo_name: str = "",
+    repo_type: Optional[str] = None,
+    prot_master: Optional[bool] = None,
+    prot_pages: Optional[bool] = None,
+    prot_default: Optional[bool] = None,
+    status_checks: Optional[str] = None,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    if repo_data_path:
+        # run for each repo in list
+        repo_data = parse_repo_list_properties(repo_data_path)
+        for repo_prop in repo_data:
+            repo_prop.apply_properties(write=write)
+        return
+
+    # Apply properties for a single repo
+    if not repo_name:
+        return
+
+    if repo_type is not None:
+        repo_type = RepoType(repo_type)
+    if status_checks is not None:
+        status_checks = StatusChecks(status_checks)
+
+    prop = RepoProperty(
+        owner=owner, repo_name=repo_name, repo_type=repo_type,
+        master=prot_master, default=prot_default,
+        gh_pages=prot_pages,
+        status_checks=status_checks
+    )
+
+    prop.apply_properties(write=write)
+
+
+def _create_argparser() -> argparse.ArgumentParser:
+    """
+    Create an ArgumentParser for update_repo_properties
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    # specify an owner, repo, and settings
+    parser.add_argument("owner", type=str, default='pcdshub', nargs='?',
+                        help='Organization or owner of the repository, "pcdshub" '
+                             'by default')
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("--repo_type", type=str, nargs='?',
+                        choices=list(t.value for t in RepoType),
+                        help='Type of the repository. (Default: Other)')
+    parser.add_argument("--status_checks", type=str, nargs='?',
+                        choices=list(t.value for t in StatusChecks),
+                        help="Status checks to require before merging a Pull Request."
+                             " (Default: None)")
+    parser.add_argument("--protect-master", action=argparse.BooleanOptionalAction,
```

Comment:
One more thing, just a comprehension question:
Does this mean:

- no arg provided -> no change
-  `--protect-master` provided -> set to `True`
- `--no-protect-master` provided -> set to `False`

Reply:
That's correct. If the flag isn't supplied, the argument is `None`